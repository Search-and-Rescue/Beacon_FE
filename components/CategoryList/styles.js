import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  categoryContainer: {
    marginTop: 80,
    backgroundColor: "red"
  },
  categoryList: {

  },
  addItemBtn: {
    fontSize: 21,
    height: 50,
    textAlign: 'center'
  }
});

export default styles;