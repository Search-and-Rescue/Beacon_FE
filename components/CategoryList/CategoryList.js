import React, { Component } from 'react';
import { View, Text, Button } from 'react-native';
import styles from './styles';
// import MenuButton from "../../navigation/MenuButton/MenuButton";

// const CategoryList = (props) => {
//   const { routeName } = props.navigation.state;

export default class CategoryList extends Component {
  constructor(props) {
    super(props)
  }

  render() {
  return (
    <View>
      <View style={styles.categoryContainer}>
        <View style={styles.categoryList}></View>
        <Text>Category List Page</Text>
        <View style={styles.listItemBtn}>
          <Button
            title="Add Item"
            style={styles.addItemBtn}
            onPress={() => this.props.navigation.navigate("ListItem")}
          />
        </View>
      </View>
    </View>
  );
}
}