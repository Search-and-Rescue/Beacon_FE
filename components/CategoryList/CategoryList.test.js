import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import CategoryList from './CategoryList';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

jest.mock('../../navigation/MenuButton/MenuButton', () => 'MenuButton');

const categoryMock = {
  navigation: {
    state: {
      routeName: 'Profile'
    }
  }
}

describe('CategoryList', () => {
  it('should match the snapshot with all of the data passed through', () => {
    expect(shallow(<CategoryList {...categoryMock}/>)).toMatchSnapshot();
  });
});
