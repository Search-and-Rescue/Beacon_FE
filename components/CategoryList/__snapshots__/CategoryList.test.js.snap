// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList should match the snapshot with all of the data passed through 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "red",
        "marginTop": 80,
      }
    }
  >
    <View
      style={Object {}}
    />
    <Text>
      Category List Page
    </Text>
    <View>
      <Button
        onPress={[Function]}
        style={
          Object {
            "fontSize": 21,
            "height": 50,
            "textAlign": "center",
          }
        }
        title="Add Item"
      />
    </View>
  </View>
</View>
`;
